src/Tribe/DB_Lock.php

```diff
@@ -0,0 +1,355 @@
+	 * Acquires a db lock.
+	 *
+	 * To ensure back-compatibility with MySQL 5.6, the lock will hash the lock key using SHA1.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The name of the db lock key to acquire.
+	 *
+	 * @return bool Whether the lock acquisition was successful or not.
+	 */
+	public function acquire_db_lock( $lock_key ) {
+		/**
+		 * Filters the timeout, in seconds, of the database lock acquisition attempts.
+		 *
+		 * The timeout will not be used when locks are managed using queries in place of
+		 * MySQL functions.
+		 *
+		 * @since TBD
+		 *
+		 * @param int    $timeout  The timeout, in seconds, of the lock acquisition attempt.
+		 * @param string $lock_key The lock key the target of the acquisition attempt.
+		 * @param static $this     The object that's trying to acquire the lock by means of the trait.
+		 */
+		$timeout = apply_filters( 'tribe_db_lock_timeout', 3, $lock_key, $this );
+
+		if ( $this->manage_db_lock_w_mysql_functions() ) {
+			return $this->acquire_db_lock_w_mysql_functions( $lock_key, $timeout );
+		}
+
+		return $this->acquire_db_lock_w_queries( $lock_key );
+	}
+
+	/**
+	 * Returns whether the traits should try to acquire and release locks using MySQL `GET_LOCK` and `RELEASE_LOCK`
+	 * functions or not.
+	 *
+	 * If not, then the trait will manage the locks by means of direct SQL queries on the options table.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether the trait should use MySQL functions to manage the locks, or not.
+	 */
+	protected function manage_db_lock_w_mysql_functions() {
+		/**
+		 * Filters whether the database lock should be acquired using the `GET_LOCK` and `RELEASE_LOCK`
+		 * MySQL functions or not.
+		 *
+		 * If the filter returns a falsy value, then the trait will attempt to manage locks using `SELECT`
+		 * and `UPDATE` queries on the options table.
+		 *
+		 * @since TBD
+		 */
+		return tribe_is_truthy( apply_filters( 'tribe_db_lock_use_msyql_functions', true ) );
+	}
+
+	/**
+	 * Tries to acquire the database lock using MySQL functions (`GET_LOCK` and `IS_FREE_LOCK`).
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The lock key to try and acquire the lock for.
+	 * @param int    $timeout  The timeout, in seconds, to try and acquire the lock.
+	 *
+	 * @return bool Whether the lock was acquired or not.
+	 */
+	protected function acquire_db_lock_w_mysql_functions( $lock_key, $timeout ) {
+		/*
+		 * On MySQL 5.6 if a session (a db connection) fires two requests of `GET_LOCK`, the lock is
+		 * implicitly released and re-acquired.
+		 * While this will not cause issues in the context of different db sessions (e.g. two diff. PHP
+		 * processes competing for a lock), it would cause issues when the lock acquisition is attempted
+		 * in the context of the same PHP process.
+		 * To avoid a read-what-you-write issue in the context of the same request, we check if the lock is
+		 * free, using `IS_FREE_LOCK` first.
+		 */
+
+		global $wpdb;
+
+		// phpcs:ignore
```

Comment:
Why Ignore?

Reply:
`get_var` calls should, traditionally, be accompained by calls to invalidate/update cache; there is no caching component here, so I've suppressed them.